docs/web/howto/client.rst

```diff
@@ -331,74 +331,85 @@ This function returns a ``Deferred`` that fires with the body after the request
 
 
 
-HTTP over SSL
-~~~~~~~~~~~~~
-
-
-    
-
-Everything you've read so far applies whether the scheme of the request
-URI is *HTTP* or *HTTPS* .  However, to control the SSL
-negotiation performed when an *HTTPS* URI is requested, there's
-one extra object to pay attention to: the SSL context factory.
-
-
-
-
-
+Customizing your HTTPS Configuration
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-``Agent`` 's constructor takes an optional second argument, a context factory.
-Generally you don't want to create the context factory yourself: it's easy to get wrong.
-Instead, Twisted provides helpers for building correctly configured ones.
-For example, you can use :api:`twisted.internet.ssl.optionsForClientTLS <optionsForClientTLS>`, which takes care of all hostname certificate verification for you.
+Everything you've read so far applies whether the scheme of the request URI is *HTTP* or *HTTPS* .
 
+By default, since version 15.0.0 (if you're using an earlier version, please upgrade, as this is a critical security feature!), Twisted will automatically validate ``https`` URLs against your platform's trust store to the best of its ability.
+Be sure to ``pip install twisted[tls]`` in order to get all the dependencies necessary to do TLS properly, but beyond that you should not need to do much.
 
+However, for some uses, you may need to customize Agent's use of HTTPS; for example, to provide a client certificate, or to use a custom certificate authority for an internal network.
 
+Here, we're just going to show you how to inject the relevant TLS configuration into an Agent, so we'll use the simplest possible example, rather than a more useful but complex one.
 
+``Agent``'s constructor takes an optional second argument, which allows you to customize its behavior with respect to HTTPS.
```

Comment:
> My reading of the statement is that it is a restrictive clause. The context of the sentence makes the HTTPS customization the critical information being conveyed here. Maybe that actually means the sentence should be something more like "You can customize the behavior of Agent with respect to HTTPS by using its optional second argument." and that's the edit I should have suggested rather than just the comma removal (since technically "Agent's constructor takes an optional second argument." makes perfect sense on its own).

My understanding of "restrictive" is that, while it is often colloquially described as "provides important information", what it actually means is that it provides _essential_ information, i.e. information without which the meaning would be _different_.  If you can provide the additional information in separate sentences, then the clause, while it might be important, is not restrictive.

To phrase this in terms of the things we're talking about here, in this case the clause would be restrictive:

>> Agent's constructor takes one argument which allows you to customize its behavior with respect to HTTPS.

but non-restrictive in this case:

>> Agent's constructor takes a second argument, which allows you to customize its behavior with respect to HTTPS.

since in the first case if you eliminate the "which" clause, the sentence could legitimately be interpreted as "Agent's constructor takes _only_ one argument", whereas in the second case you could write it as: 

>> Agent's constructor takes a second argument.  It allows you to customize its behavior with respect to HTTPS.